### PR TITLE
ReviewController fixes

### PR DESCRIPTION
**Pull request by Berg1989**
ReviewController now works with our recent changes.

After posting a review, the controller will redirect to catalog/product url, even is there should be a currentCategory. This has been added to our spreadsheet of tasks.

### CODE REVIEW

File: src/Ucommerce.Avenue.Umbraco/Views/PartialView/ProductReview.cshtml [hunk 1]
```diff
@@ -102,7 +102,7 @@ using (Html.BeginUmbracoForm("Index", "Review"))
             <div class="control-group margin-top">
                 <div class="controls">
                     <button type="submit" class="btn btn-primary" name="review-product">Save</button>
-                    <input type="hidden" name="reviewed-product-sku" id="@product.Sku" />
+                    <input type="hidden" id="reviewed-product-sku" name="ProductSku" value="@product.Sku" />
```
Comment (ilny) [inline src/Ucommerce.Avenue.Umbraco/Views/PartialView/ProductReview.cshtml]: Not sure about this... Please make sure it doesn't break anything. Is changing this even necessary?

Comment (Berg1989) [reply src/Ucommerce.Avenue.Umbraco/Views/PartialView/ProductReview.cshtml]: This change is what made it possible to grab the Sku in the controller. I think this has always been a mistake, but it just wasn't used to anything, so it wasn't discovered until now.

File: src/Ucommerce.Avenue.Umbraco/Controllers/ReviewController.cs [hunk 2]
```diff
@@ -56,16 +56,21 @@ private IList<ProductReviewViewModel> MapReviews(Product product)
...
         [HttpPost]
         public ActionResult Index(ProductReviewViewModel formReview)
         {
-            var product = CatalogContext.CurrentProduct;
-            var category = CatalogContext.CurrentCategory;
+            var product = CatalogLibrary.GetProduct(formReview.ProductSku);
+            var productV2 = Product.FirstOrDefault(x => x.Guid == product.Guid); 
+
+            // var category = CatalogContext.CurrentCategory;
+            //  var categoryV2 = Category.FirstOrDefault(x => x.Guid == category.Guid);
```
[the remark this answers is not in the document]
Comment (Berg1989) [reply src/Ucommerce.Avenue.Umbraco/Controllers/ReviewController.cs]: This as well as the TODO below this was just meant to make it easier for whoever picks it up next time to know where to look, but I can remove it if necessary :)

[the remark this answers is not in the document]
Comment (ilny) [reply src/Ucommerce.Avenue.Umbraco/Controllers/ReviewController.cs]: That's a good point. Nice foundation but not quite there yet as you mentioned, let's stick `// TODO:` on line 68 and revisit it later, unles @skovsboll @vitalie-ucommerce have any objections.

File: src/Ucommerce.Avenue.Umbraco/Views/PartialView/ProductReview.cshtml (hunk 1 above)
Comment (ilny) [reply src/Ucommerce.Avenue.Umbraco/Views/PartialView/ProductReview.cshtml]: Good catch!

Comment (Berg1989) [reply src/Ucommerce.Avenue.Umbraco/Views/PartialView/ProductReview.cshtml]: At least half the credit goes to @skovsboll  :p But thanks!

**Pull request merged**